### ISSUE TITLE
Should Timestamps have to be in order?

### ISSUE THREAD

**Issue by ianswett** [wcsmith/draft-quic-receive-ts#2]
Currently timestamps must decrease in value as the packet numbers decrease in value, so if a packet with a lower packet number is received after a packet with a higher packet number, that timestamp can't be expressed in the frame.

For existing CC algorithms which use timestamps, it's believed this is ok, but it'd be good to confirm that design decision.

**Comment by ianswett** [comment]
@jbeshay proposed a way to do this at IETF 118: https://datatracker.ietf.org/meeting/118/materials/slides-118-quic-ack-timestamps-00.pdf

So far, there's substantial support for this, so I think it's worth writing a PR.

**Comment by jbeshay** [comment]
Yes, @sharadjaiswal gave that presentation and he had added it to the mvfst implementation. We'll work on a PR to update the draft soon.

**Comment by chris-box** [comment]
Thumbs up for the above. I'd like to see this happen due to the routine layer 2 reordering described by https://gwhitecl.github.io/draft-white-intarea-reordering/draft-white-intarea-reordering.html. Widespread adoption of support for out-of-order timestamps is then an enabler for layer 2s to be less aggressive with imposing transmission delays while they put packets back in the original sequence.

**Comment by ianswett** [comment]
This was fixed by #15
